app/controllers/locations_controller.rb

```diff
@@ -8,6 +8,7 @@ def new
   def create
     @location = Location.new(location_params)
     if @location.save
+      @stocks = Product.all.each { |product| Stock.create!(location: @location, product: product, quantity: 0) }
```

Comment:
`@stocks` instance variable is not used anywhere, we can skip the assignment. 

It would be also good to run these operations inside a transaction (both @location.save and creating stocks). 

You can read more about transactions and why are they important here: https://medium.com/@kristenrogers.kr75/rails-transactions-the-complete-guide-7b5c00c604fc